app/models/user.rb

```diff
@@ -178,6 +178,14 @@ def direct_filters
   has_many :log_items, :dependent => :destroy
   validates_associated :log_items
 
+  after_update :expire_caches
+
+  def expire_caches
+    if login_changed?
+      self.works.each{ |work| work.touch }
```

Comment:
Redundant `self` detected.<br>Space missing to the left of {.<br>Pass `&:touch` as an argument to `each` instead of a block.